### PR TITLE
Fix NPEs cause by null times

### PR DESCRIPTION
**Pull request by alexabird**

### CODE REVIEW

File: src/main/java/org/codice/itest/TestExecutorTask.java [hunk 1]
```diff
@@ -59,19 +59,15 @@ public void run() {
 
             this.notify(testResultFactory.pass(testName, beforeTest, afterTest));
         } catch (AssertionError e) {
-            if(beforeTest!=null && afterTest==null)
-                afterTest = Instant.now();
```
Comment (dcruver) [inline src/main/java/org/codice/itest/TestExecutorTask.java:63]: :pencil: We could declare this in `notify()` and drop the conditional.

File: src/main/java/org/codice/itest/TestExecutorTask.java [hunk 2]
```diff
@@ -59,19 +59,15 @@ public void run() {
...
+            afterTest = Instant.now();
             this.notify(testResultFactory.fail(testName, e.getMessage(), beforeTest, afterTest));
         } catch (Throwable t) {
-            if(beforeTest!=null && afterTest==null)
-                afterTest = Instant.now();
```
Comment (dcruver) [inline src/main/java/org/codice/itest/TestExecutorTask.java:67]: :memo:  We could declare this in `notify()` and drop the conditional.

File: src/main/java/org/codice/itest/TestExecutorTask.java [hunk 3]
```diff
@@ -59,19 +59,15 @@ public void run() {
...
+            afterTest = Instant.now();
             this.notify(testResultFactory.error(testName, t, beforeTest, afterTest));
         } finally {
             try {
                 diagnosticTest.cleanup();
             } catch (Exception e) {
-                if(beforeTest!=null && afterTest==null)
-                    afterTest = Instant.now();
```
Comment (dcruver) [inline src/main/java/org/codice/itest/TestExecutorTask.java:74]: :memo:  We could declare this in `notify()` and drop the conditional.

**Review by dcruver — COMMENTED** [review]
:exclamation: Even with this change in place, if `setup()` throws an exception, we're going to get an NPE. I'd suggest we switch lines 55 and 56 and move all occurrences of `afterTest = Instant.now()` into `notify()`. Then you can remove the conditionals as well.

**Comment by alexabird** [comment]
@dcruver I made some changes that make this cleaner and I think address your core concerns. 
* Called `Instant.now()` for the endTime paramter in the `TestResult` initializers, instead of moving `afterTest = Instant.now()` to the `notify()` method (would require more changes than just that line move since the `TestResult` initializers need the `endTime`)
* Moved `setup()` within the timed part of the test, and also moved `cleanup()` into timed portion.

**Review by mcalcote — APPROVED** [review]

**Review by dcruver — APPROVED** [review]

**Pull request merged**